packages/ingest/src/granule.ts

```diff
@@ -361,149 +362,74 @@ export function generateMoveFileParams(
 * @param {MoveFileParams} moveFileParam - Parameter object describing the move operation
 * @param {FilePgModel} filesPgModel - FilePgModel instance
 * @param {Knex.Transaction | Knex} trx - Knex transaction or (optionally) Knex object
 * @param {number | undefined } postgresCumulusGranuleId - postgres internal granule id
-* @param {boolean} writeToPostgres - explicit flag to enable/disable postgres database updates
-* @returns {ReturnValueDataTypeHere} Brief description of the returning value here.
+* @returns {MovedGranuleFile}
 */
 export async function moveGranuleFile(
   moveFileParam: MoveFileParams,
   filesPgModel: FilePgModel,
   trx: Knex.Transaction | Knex,
-  postgresCumulusGranuleId: number | undefined,
-  writeToPostgres: boolean = true
-): Promise<MovedGranuleFile> {
+  postgresCumulusGranuleId: number | undefined
+): Promise<Omit<ApiFile, 'granuleId'>> {
   const { source, target, file } = moveFileParam;
-  if (moveFileParam.source && moveFileParam.target) {
+
+  if (source && target) {
     log.debug('moveGranuleS3Object', source, target);
-    if (writeToPostgres) {
-      if (!postgresCumulusGranuleId) {
-        throw new Error('postgresCumulusGranuleId must be defined to move granule file if writeToPostgres is true');
-      }
-      const cumulusId = await filesPgModel.getRecordCumulusId(trx, {
+    if (!postgresCumulusGranuleId) {
+      throw new Error('postgresCumulusGranuleId must be defined to move granule file if writeToPostgres is true');
+    }
+    const updatedPgRecords = await filesPgModel.update(
+      trx,
+      {
         granule_cumulus_id: postgresCumulusGranuleId,
-        bucket: moveFileParam.source.Bucket,
-        key: moveFileParam.source.Key,
-      });
-      await filesPgModel.update(trx, {
-        cumulus_id: cumulusId,
+        bucket: source.Bucket,
+        key: source.Key,
       },
       {
-        bucket: moveFileParam.target.Bucket,
-        key: moveFileParam.target.Key,
+        bucket: target.Bucket,
+        key: target.Key,
         file_name: getNameOfFile(file),
-      });
+      }, ['*']
+    );
+    if (updatedPgRecords.length !== 1) {
+      throw new RecordDoesNotExist('Attempted to update granule on move, but granule does not exist');
     }
+    const updatedPgRecord = updatedPgRecords[0];
+
     await S3.moveObject({
-      sourceBucket: moveFileParam.source.Bucket,
-      sourceKey: moveFileParam.source.Key,
-      destinationBucket: moveFileParam.target.Bucket,
-      destinationKey: moveFileParam.target.Key,
+      sourceBucket: source.Bucket,
+      sourceKey: source.Key,
+      destinationBucket: target.Bucket,
+      destinationKey: target.Key,
       copyTags: true,
     });
 
     return {
-      bucket: moveFileParam.target.Bucket,
-      key: moveFileParam.target.Key,
-      name: getNameOfFile(file),
+      ...translatePostgresFileToApiFile(updatedPgRecord),
+      bucket: target.Bucket,
+      key: target.Key,
+      fileName: getNameOfFile({ key: target.Key }),
```

Comment:
Yes, this could be 

```
return translatePostgresFileToApiFile({...updatedPgRecord, bucket: target.Bucket, key:target.Key, filename: getNameOfFile({ key: target.Key }));
```

although the explicit return typing on the method also explicitly prevents what you're talking about.